src/openrct2-ui/windows/SceneryScatter.cpp

```diff
@@ -0,0 +1,238 @@
+        | (1 << WIDX_DENSITY_LOW) | (1 << WIDX_DENSITY_MEDIUM) | (1 << WIDX_DENSITY_HIGH);
+    window->hold_down_widgets = (1 << WIDX_INCREMENT) | (1 << WIDX_DECREMENT);
+    window_init_scroll_widgets(window);
+    window_push_others_below(window);
+
+    return window;
+}
+
+static void window_scenery_scatter_mouseup(rct_window* w, rct_widgetindex widgetIndex)
+{
+    switch (widgetIndex)
+    {
+        case WIDX_CLOSE:
+            gWindowSceneryScatterEnabled = 0;
+            window_close(w);
+            break;
+        case WIDX_PREVIEW:
+            window_scenery_scatter_inputsize(w, widgetIndex);
+            break;
+
+        case WIDX_DENSITY_LOW:
+            gWindowSceneryScatterDensity = ScatterToolDensity::LowDensity;
+            break;
+
+        case WIDX_DENSITY_MEDIUM:
+            gWindowSceneryScatterDensity = ScatterToolDensity::MediumDensity;
+            break;
+
+        case WIDX_DENSITY_HIGH:
+            gWindowSceneryScatterDensity = ScatterToolDensity::HighDensity;
+            break;
+    }
+}
+
+static void window_scenery_scatter_mousedown(rct_window* w, rct_widgetindex widgetIndex, [[maybe_unused]] rct_widget* widget)
+{
+    switch (widgetIndex)
+    {
+        case WIDX_DECREMENT:
+            // Decrement land tool size, if it stays within the limit
+            gWindowSceneryScatterSize = std::max(MINIMUM_TOOL_SIZE, gWindowSceneryScatterSize - 1);
+
+            // Invalidate the window
+            w->Invalidate();
+            break;
+        case WIDX_INCREMENT:
+            // Increment land tool size, if it stays within the limit
+            gWindowSceneryScatterSize = std::min(MAXIMUM_TOOL_SIZE, gWindowSceneryScatterSize + 1);
+
+            // Invalidate the window
+            w->Invalidate();
+            break;
+    }
+}
+
+static void window_scenery_scatter_textinput(rct_window* w, rct_widgetindex widgetIndex, char* text)
+{
+    int32_t size;
+    char* end;
+
+    if (widgetIndex != WIDX_PREVIEW || text == nullptr)
+        return;
+
+    size = strtol(text, &end, 10);
+    if (*end == '\0')
+    {
+        switch (widgetIndex)
+        {
+            case WIDX_PREVIEW:
+                size = std::max(MINIMUM_TOOL_SIZE, size);
+                size = std::min(MAXIMUM_TOOL_SIZE, size);
+                gWindowSceneryScatterSize = size;
+                break;
+        }
+        w->Invalidate();
+    }
+}
+
+static void window_scenery_scatter_inputsize(rct_window* w, rct_widgetindex widgetindex)
+{
+    uint8_t maxlen = 0;
+    switch (widgetindex)
+    {
+        case WIDX_PREVIEW:
+            TextInputDescriptionArgs[0] = MINIMUM_TOOL_SIZE;
+            TextInputDescriptionArgs[1] = MAXIMUM_TOOL_SIZE;
+            maxlen = 3;
+            break;
+    }
+    window_text_input_open(w, widgetindex, STR_SELECTION_SIZE, STR_ENTER_SELECTION_SIZE, STR_NONE, STR_NONE, maxlen);
+}
+
+static void window_scenery_scatter_update(rct_window* w)
+{
+    w->frame_no++;
```

Comment:
I think we can safely remove this event as well, as the window does not use any animations. (I already removed the empty `close` event.)